lib/colmena/cell.rb

```diff
@@ -0,0 +1,68 @@
+module Colmena
+  module Cell
+    module ClassMethods
+      def ports
+        @ports ||= []
+      end
+
+      def register_port(port)
+        ports << port
+      end
+
+      def commands
+        @commands ||= {}
+      end
+
+      def register_command(command)
+        commands[class_to_sym(command)] = command
+      end
+
+      def queries
+        @queries ||= {}
+      end
+
+      def register_query(query)
+        queries[class_to_sym(query)] = query
+      end
+
+      def self.class_to_sym(klass)
+        name_without_namespace = klass.name.split('::').last
+        name_without_namespace.gsub(/([^\^])([A-Z])/,'\1_\2').downcase.to_sym
+      end
+    end
+
+    def self.included(klass)
+      klass.extend ClassMethods
+    end
+
+    def initialize(adapters={})
+      @commands = inject_ports(self.class.commands)
```

Comment:
We need to initialize the `queries` here too!

```suggestion
      @commands = inject_ports(self.class.commands)
      @queries = inject_ports(self.class.queries)
```

Reply:
I already did this in #17.

Reply:
gr8, thank you!